sympy/interactive/printing.py

```diff
@@ -485,7 +485,7 @@ def init_printing(pretty_print=True, order=None, use_unicode=None,
 
     if pretty_print:
         if pretty_printer is not None:
-            stringify_func = pretty_printer
+            stringify_func = lambda expr: pretty_printer(expr,**settings)
```

Comment:
I think you are introducing a couple of errors:

1. You are passing `settings` to `pretty_printer`. But at this stage of execution, `settings` may contains keyword arguments not supported by `pretty_printer`.
2. You applied an edit to the body of an `if` statement, but you didn't look at what happens in the body of the `else` statement. In this particular case, the `else` statement deals with custom pretty printers. So, the edit you committed is not going to fix the issue.

I'll suggest to revert this edit. Then, take all the time you need to understand the code before attempting to a new commit. Don't be scared, it might take a while but you'll get there :)

Reply:
Thank you @Davide-sd for the guidance, I will close this PR and will try to understand more about the code.